src/lightning/pytorch/loops/evaluation_loop.py

```diff
@@ -293,13 +294,21 @@ def _on_evaluation_model_eval(self) -> None:
         """Sets model to eval mode."""
         trainer = self.trainer
         hook_name = "on_test_model_eval" if trainer.testing else "on_validation_model_eval"
-        call._call_lightning_module_hook(trainer, hook_name)
+        if is_overridden(hook_name, trainer.lightning_module):
+            call._call_lightning_module_hook(trainer, hook_name)
+        else:
+            self._module_mode.capture(trainer.lightning_module)
+            assert trainer.model is not None
+            trainer.model.eval()
```

Comment:
Why don't we still call the hook unconditionally but capture the state before?

```suggestion
        self._module_mode.capture(trainer.lightning_module)
        call._call_lightning_module_hook(trainer, hook_name)
```